### DISCUSSION TITLE
This firmware with upgrades?

### DISCUSSION THREAD

**Discussion by scaramonga** [Jyers/Marlin#841] [Q&A] [answered]
Hello all.

If I edit source code for my Ender 3 V2 (4.2.7), am I still getting all the benefits of the Jyers firmware?  I have a BLTouch that I wish 5x5 probing (HS), and upgraded thermistor/heater - 50W (table 13 in Marlin, I presume?) with Micro Swiss DD & Hotend, and dual Z axis steppers.  I have Satsana remix head also (the one with 2 fans).  What is best way to achieve this?

Thanking everyone kindly.

**Comment by smooreace** [comment, marked as answer]
Jyers firmware is nothing more than standard Marlin firmware with a ton of
work don't to make the display menus intelligent and useful. So your first
question really has no answer... its still Marlin-based firmware.

The MicroSwiss DD unit doesn't involve changing the thermistor/heater, so
there are 2 answers based on 2 directions of assumption...
1 - The DD unit instructions tell you the changes that are required to be
done in the firmware. There is nothing related to the  thermistor/heater at
all.
2 - Same as above... follow the instructions from MicroSwiss. You will also
have to make the needed modifications related to the new thermistor, based
on the information the supplier gave (or should have given) you.

All of the editings in the firmware build are exactly the same no matter if
you are running stock Marlin, Jyers, or Smith3D firmware. Therefore, your
best bet, or only option really, is to compile your own firmware.

**Reply by scaramonga** [reply]
Thank you for your kind reply.

1st.  I know it's all based on 'Marlin', but I was wondering if Jyers (source) has all these great features, before/after me editing, or does that require me to load a specific conf into code?

2nd. I have taken it upon myself to replace the thermistor/heater, as that is what I wanted, and with an all metal hotend now, it's advisable for higher temps, rather than stock ones. :)

3. I know I need modifications to firmware, which is why I asked in first place, and plan to compile the firmware for myself anyway :)

4. I was just making sure I was headed in right direction, and I know all about 'offsets' etc.

5. The Micro Swiss DD is compatible with the Ender 3 V2, but one has to change hotend cover, as original does not fit, so were talking major changes, as regards BLTouch etc., which is why I'm now using the Satsana with dual Noctua fans, hence more weight, so Dual Z Axis upgrade, with the extra stepper.

So, I guess the question is, what does the 'source' contain at bear minimum?  Is it 'bog' standard Marlin, or does one have to inject Jyers before, or after making ones edits?

**Reply by Cal-LF** [reply]
If you compile just the unzipped code folder, you will get the basic "Default No Probe" firmware. I just did this recently. Copy any of the offered E3V2 templates offered in the Configuration Files to include the extras like manual mesh leveling or BLT or UBL.
